File: facade/java/Restaurant.java
package facade.java;

public class Restaurant {
    ColdFood coldFood;
    HotFood hotFood;

    Restaurant(ColdFood coldFood, HotFood hotFood) {
        this.coldFood = coldFood;
        this.hotFood = hotFood;
    }

    String operation() {
        String result = "";
        result += coldFood.washAndRinse() + "\n";
        result += coldFood.wrap() + "\n";
        result += coldFood.freeze() + "\n";
        result += hotFood.unwrap() + "\n";
        result += hotFood.clean() + "\n";
        result += hotFood.cook() + "\n";
        return result;
    }
}

File: facade/golang/restaurant.go
package main

type Restaurant struct {
	hotFood  *HotFood
	coldFood *ColdFood
}

func NewRestaurant(hotFoodName, coldFoodName string) *Restaurant {
	return &Restaurant{
		hotFood:  NewHotFood(hotFoodName),
		coldFood: NewColdFood(coldFoodName),
	}
}

func (r *Restaurant) ServeHotFood() []string {
	var results []string
	results = append(results, r.hotFood.Unwrap())
	results = append(results, r.hotFood.Clean())
	results = append(results, r.hotFood.Cook())
	results = append(results, r.hotFood.Prepare())
	results = append(results, r.hotFood.Serve())
	return results
}

func (r *Restaurant) ServeColdFood() []string {
	var results []string
	results = append(results, r.coldFood.WashAndRinse())
	results = append(results, r.coldFood.Wrap())
	results = append(results, r.coldFood.Freeze())
	results = append(results, r.coldFood.Prepare())
	results = append(results, r.coldFood.Serve())
	return results
}
